File: Part1_강의자료/Ch06_누적합/문제별코드/11659_구간합구하기4/11659.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
arr = [0] + list(map(int, input().split()))

acc = [0] * (N + 1)
for i in range(1, N + 1):
    acc[i] = acc[i - 1] + arr[i]

for _ in range(M):
    i, j = map(int, input().split())
    print(acc[j] - acc[i - 1])

File: Part1_강의자료/Ch06_누적합/문제별코드/11659_구간합구하기4/11659.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, M;

    scanf("%d %d", &N, &M);
    vector<int> arr(N + 1);
    for (int i = 1; i <= N; i++)
        scanf("%d", &arr[i]);

    vector<int> acc(N + 1);
    for (int i = 1; i <= N; i++)
        acc[i] = acc[i - 1] + arr[i];

    while (M--)
    {
        int i, j;
        scanf("%d %d", &i, &j);
        printf("%d\n", acc[j] - acc[i - 1]);
    }
    return 0;
}
